cmd/mailbox-controller/main.go

```diff
@@ -308,65 +332,65 @@ func (ctl *mbCtl) sync1(ctx context.Context, ref any) {
 	}
 }
 
-func (ctl *mbCtl) sync(ctx context.Context, ref any) bool {
+func (ctl *mbCtl) sync(ctx context.Context, refany any) bool {
 	logger := klog.FromContext(ctx)
-	wsName, ok := ref.(string)
+	ref, ok := refany.(syncTargetRef)
 	if !ok {
-		logger.Error(nil, "Sync expected a string", "ref", ref, "type", fmt.Sprintf("%T", ref))
-		return false
-	}
-	wsNameParts := strings.Split(wsName, wsNameSep)
-	if len(wsNameParts) != 2 {
-		logger.Error(nil, "Sync expected a string with 2 parts", "ref", ref, "wsNameParts", wsNameParts)
+		logger.Error(nil, "Sync expected a syncTargetRef", "ref", refany, "type", fmt.Sprintf("%T", refany))
 		return false
 	}
-	cluster := logicalcluster.Name(wsNameParts[0])
-	targetName := wsNameParts[1]
+	cluster := ref.cluster
+	targetName := ref.name
 	syncTarget, err := ctl.synctargetLister.Cluster(cluster).Get(targetName)
 	if err != nil && !k8sapierrors.IsNotFound(err) {
-		logger.Error(err, "Unable to Get referenced SyncTarget", "wsNameParts", wsNameParts)
+		logger.Error(err, "Unable to Get referenced SyncTarget", "ref", ref)
 		return true
 	}
+	wsName := ref.mailboxWSName()
 	workspace, err := ctl.workspaceLister.Get(wsName)
 	if err != nil && !k8sapierrors.IsNotFound(err) {
-		logger.Error(err, "Unable to Get referenced Workspace", "wsName", wsName)
+		logger.Error(err, "Unable to Get referenced Workspace", "ref", ref, "wsName", wsName)
 		return true
 	}
 	if syncTarget == nil || syncTarget.DeletionTimestamp != nil {
 		if workspace == nil || workspace.DeletionTimestamp != nil {
-			logger.V(3).Info("Both absent or deleting, nothing to do", "wsName", wsName)
+			logger.V(3).Info("Both absent or deleting, nothing to do", "ref", ref)
 			return false
```

Comment:
Can we explicitly say "Both syncTarget & workspace are ...."

Reply:
Done.